src/openloco/windows/CompanyFaceSelection.cpp

```diff
@@ -1,15 +1,320 @@
+    {
+        const auto objIndex = getObjectFromSelection(y);
+
+        if (!objIndex.object._header)
+        {
+            return;
+        }
+        self->invalidate();
+        audio::play_sound(audio::sound_id::click_down, _cursorX);
+        gGameCommandErrorTitle = string_ids::cant_select_face;
+        const auto result = game_commands::do_65(*objIndex.object._header, self->owner);
+        if (result)
+        {
+            WindowManager::close(self);
+        }
+    }
+
+    static void scrollMouseOver(window* const self, const int16_t x, const int16_t y, const uint8_t scroll_index)
+    {
+        const auto [rowIndex, object] = getObjectFromSelection(y);
+        if (self->row_hover == rowIndex)
+        {
+            return;
+        }
+        self->row_hover = rowIndex;
+        self->object = object._name;
+        sub_471B95();
+        if (object._header)
+        {
+            sub_47176D(*object._header);
+        }
+        self->invalidate();
+    }
+
+    // 0x4352B1
+    static void tooltip(FormatArguments& args, window* const self, const widget_index)
+    {
+        args.push(string_ids::tooltip_scroll_list);
+    }
+
+    // 0x434FE8
+    static void prepareDraw(window* const self)
+    {
+        const auto company = companymgr::get(self->owner);
+
+        FormatArguments args{};
+        args.push(company->name);
+    }
+
+    // 0x435003
+    static void draw(window* const self, gfx::drawpixelinfo_t* const dpi)
+    {
+        self->draw(dpi);
+        if (self->row_hover == -1)
+        {
+            return;
+        }
+
+        {
+            const auto colour = colour::get_shade(self->colours[1], 0);
+            const auto l = self->x + 1 + self->widgets[widx::face_frame].left;
+            const auto t = self->y + 1 + self->widgets[widx::face_frame].top;
+            const auto r = self->x - 1 + self->widgets[widx::face_frame].right;
+            const auto b = self->y - 1 + self->widgets[widx::face_frame].bottom;
+            gfx::fill_rect(dpi, l, t, r, b, colour);
+
+            const competitor_object* competitor = _loadedObject;
+            uint32_t img = competitor->images[0] + 1 + (1 << 29);
+            gfx::draw_image(dpi, l, t, img);
+        }
+
+        {
+            const auto x = self->x + self->widgets[widx::face_frame].mid_x();
+            const auto y = self->y + self->widgets[widx::face_frame].bottom + 3;
+            const auto width = self->width - self->widgets[widx::scrollview].right - 6;
+            auto str = const_cast<char*>(stringmgr::get_string(string_ids::buffer_2039));
+            *str++ = control_codes::window_colour_2;
+            strcpy(str, self->object);
+            gfx::draw_string_centred_clipped(*dpi, x, y, width, colour::black, string_ids::buffer_2039);
+        }
+
+        // There was code for displaying competitotor stats if window opened with none
+        // playing company. But that ability is disabled from the company window.
```

Comment:
This is a bit odd. Is it really dead code? The scenario editor doesn't appear to support editing companies, otherwise that'd be my guess. Could it be exposed through multiplayer mode? (Are spectators supported?)

Reply:
You can only get to this window from the company window. And the company window only enables this button when it's a player company. So I'm pretty confident you can never get here. Scenario editor as far as i could tell can't set competitors.